### DISCUSSION TITLE
[v2] Can v1-style markdown components still be used?

### DISCUSSION THREAD

**Discussion by teegee** [nuxt/content#1363] [Q&A] [answered]
I'm in the process of converting my blog to Nuxt V3 and Content V2. My articles use components in markdown like this:

```
<c-gallery title="As received" height="15rem"
           :images="[['./seiko-front.jpg', ''],
                     ['./seiko-bottom.jpg', ''],
                     ['./seiko-top.jpg', '']]">
</c-gallery>

```
V2 complains about the images prop:

```
[Vue warn]: Invalid prop: type check failed for prop "images". Expected Array, got String with value "[['./seiko-front.jpg', ''],
                  ['./seiko-bottom.jpg', ''],
                  ['./seiko-top.jpg', '']]".

```
For V1, a similar error happens when you omit the colon before 'images'. Am I missing something, or do I have to change all my articles to the new content component format?

**Comment by farnabaz** [comment, marked as answer]
Sadly this is not possible in V2. Content V1 used `eval` and `new Function` to evaluate these syntaxes, but due to security reasons, Content V2 does not use them.

However, there are alternative ways to pass props to components,

### With frontmatter
```
---
images:
- - "./seiko-front.jpg"
  - ''
- - "./seiko-bottom.jpg"
  - ''
- - "./seiko-top.jpg"
  - ''
  -
---
<c-gallery title="As received" height="15rem" :images="images">
</c-gallery>
```

### With MDC syntax and scoped variable

```

::c-gallery
---
title: As received
height: 15rem
images:
- - "./seiko-front.jpg"
  - ''
- - "./seiko-bottom.jpg"
  - ''
- - "./seiko-top.jpg"
  - ''
  -
---
::
```

**Reply by teegee** [reply]
OK, thanks for explaining. It doesn't look very pretty, but the V1 version also wasn't that great.. I guess I could simplify it to a single list, and parse the entries into a picture and a label string (empty quotes in my example), like this:

```
::c-gallery
---
title: As received
height: 15rem
images:
- "./seiko-front.jpg (Front)"
- "./seiko-bottom.jpg (Bottom)"
- "./seiko-top.jpg (Top)"
---
::
```
